### PR TITLE
latest beagle-lib support foss2016a

### PR DESCRIPTION
**Pull request by ysagon**

### CODE REVIEW

**Comment by ysagon** [comment]
yes

**Comment by boegel** [comment]
Jenkins: ok to test

(@ysagon: I don't mind you talking to our bot, but it'll only listen to certain 'admins' ;-))

File: easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'ConfigureMake'
+
+name = 'beagle-lib'
+version = '2_1_2'
```
Comment (boegel) [inline easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb]: this is a very unnatural looking version number...

please use `version = '2.1.2'`, and derive the version to be used in the source filename below

File: easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+beagle-lib is a high-performance library that can perform the core
+calculations at the heart of most Bayesian and Maximum Likelihood
+phylogenetics packages.
+"""
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+# https://github.com/beagle-dev/beagle-lib/archive/beagle_release_2_1_2.tar.gz
+source_urls = ['https://github.com/beagle-dev/beagle-lib/archive/']
+sources = ['beagle_release_%(version)s.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb]: w.r.t. comment above: please use:

```python
sources = ['beagle_release_%s.tar.gz' % version.replace('.', '_')]
```

File: easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+easyblock = 'ConfigureMake'
+
+name = 'beagle-lib'
+version = '2_1_2'
+
+homepage = 'http://code.google.com/p/beagle-lib/'
```
Comment (boegel) [inline easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb]: please change to github URL?

File: easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+phylogenetics packages.
+"""
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+# https://github.com/beagle-dev/beagle-lib/archive/beagle_release_2_1_2.tar.gz
+source_urls = ['https://github.com/beagle-dev/beagle-lib/archive/']
+sources = ['beagle_release_%(version)s.tar.gz']
+
+dependencies = [('Java', '1.7.0_15', '', True)]
```
Comment (boegel) [inline easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb]: this is very old, if you want to stick to Java 1.7, please use the latest version (1.7.0_80)

File: easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+# parallel build does not work (to test)
+parallel = 1
+
+preconfigopts = "./autogen.sh && "
```
Comment (boegel) [inline easybuild/easyconfigs/b/beagle-lib/beagle-lib-2_1_2-foss-2016a.eb]: this implies a build dependency on `Autotools`, please include this above (right near `dependencies`):

```python
builddependencies = [('Autotools', '20150215')]
```

**Comment by boegel** [comment]
@ysagon please also remove `beagle-lib-2_1_2-foss-2016a.eb`?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node2726.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/728e2f21d1ea703795e56fd3724593be for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2020.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/869b61c730045c70a2376626ab41dad9 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @ysagon!

**Pull request merged**